test_ros2cli/CMakeLists.txt

```diff
@@ -52,17 +73,40 @@ if(BUILD_TESTING)
   endmacro()
 
   set(generated_python_files)
-  macro(tests)
-    set(TEST_ROS2ACTION_DELAY 0.0)
-    if("${rmw_implementation}" STREQUAL "rmw_connext_cpp")
-      # Connext startup is too slow. It needs a few seconds until discovery starts working.
-      set(TEST_ROS2ACTION_DELAY 5.0)
-    endif()
-    custom_test_process_output(test_ros2action config_ros2action_test 180 ${TEST_ROS2ACTION_DELAY})
-    custom_test_process_output(test_ros2msg config_ros2msg_test 60 0.0)
+  macro(custom_cli_tests)
+    add_custom_cli_test(
+      test_ros2action
+      CONFIG_MODULE ros2action_test_configuration
+      TIMEOUT 300)
+    add_custom_cli_test(
+      test_ros2topic
+      CONFIG_MODULE ros2topic_test_configuration
+      TIMEOUT 300)
+    add_custom_cli_test(
+      test_ros2service
+      CONFIG_MODULE ros2service_test_configuration
+      TIMEOUT 300)
+    add_custom_cli_test(
+      test_ros2node
+      CONFIG_MODULE ros2node_test_configuration
+      TIMEOUT 240)
+    add_custom_cli_test(
+      test_ros2msg
+      CONFIG_MODULE ros2msg_test_configuration
+      TIMEOUT 240)
+    add_custom_cli_test(
+      test_ros2srv
+      CONFIG_MODULE ros2srv_test_configuration
+      TIMEOUT 240)
+    add_custom_cli_test(
+      test_ros2interface
+      CONFIG_MODULE ros2interface_test_configuration
+      TIMEOUT 240)
```

Comment:
The sum of all timeouts is 31 min. Is that really necessary?

Reply:
It's insane. I was going to bring it up in tomorrow's meeting. And yes, it used to be half of it and tests would timeout in CI. Cutting down delays we may be able to reduce it, but I wouldn't expect a large reduction e.g. Connext is **very** slow.